### PR TITLE
Report-only CSP + minor docs and cleanup updates

### PR DESCRIPTION
**Pull request by jezdez**

### CODE REVIEW

**Comment by jonparrott** [comment]
@jezdez thanks for this! I would've preferred to separate PRs but this is fine.

One reservation - the CSP report-only mode requires a report-uri, but this doesn't provide any way to set that. Do you want to add that in?

**Comment by jezdez** [comment]
Ah, I'll split the commits in separate PRs next time, apologies.

The [report-uri](https://developer.mozilla.org/en-US/docs/Web/Security/CSP/CSP_policy_directives#report-uri) directive basically works like all the other directives, so either we can:

1. Only document that a report URI needs to be set as part of the `content_security_policy` (e.g. `content_security_policy="default-src 'self'; report-uri: https://report-uri.io/example"`)
2. require the new `content_security_policy_report_only` to take a string that auto-sets the `content_security_policy['report-uri']` value
or
3. add a separate config parameter `content_security_policy_report_uri` just to be explicit

Not sure what you prefer, I'm happy to do the implementation

**Comment by jonparrott** [comment]
I vote for (3). Thank you!

**Comment by jezdez** [comment]
@jonparrott Done! Sadly I had to add an exception handling on `Talisman.__init__` when `content_security_policy_report_only` is set but no `content_security_policy_report_uri` is given. Unless you can think of a better way?

File: README.rst [hunk 1]
```diff
@@ -76,7 +85,14 @@ Options
 -  ``strict_transport_security_include_subdomains``, default ``True``,
    whether subdomains should also use HSTS.
 -  ``content_security_policy``, default ``default-src: 'self'``, see the
-   `section below <#content-security-policy>`__.
+   `section below <#content-security-policy>`_.
+-  ``content_security_policy_report_only``, default ``True``, whether to set
```
Comment (jonparrott) [inline README.rst]: The default is `False`

File: README.rst [hunk 2]
```diff
@@ -76,7 +85,14 @@ Options
...
+    the CSP header as "report-only" (as `Content-Security-Policy-Report-Only`)
+    to ease deployment by disabling the policy enforcement by the browser,
+    requires ``report-uri`` parameter passed to ``content_security_policy``
```
Comment (jonparrott) [inline README.rst]: The parameter is `content_security_policy_report_uri` not `report-uri`, right?

File: talisman/talisman.py [hunk 3]
```diff
@@ -101,12 +109,21 @@ def init_app(
...
         self.content_security_policy = content_security_policy.copy()
+        self.content_security_policy_report_uri = \
+            content_security_policy_report_uri
+        self.content_security_policy_report_only = \
+            content_security_policy_report_only
+        if self.content_security_policy_report_only and \
+                self.content_security_policy_report_uri is None:
+            raise ValueError(
+                'The parameter content_security_policy_report_only '
+                'reqires setting content_security_policy_report_uri, too.')
```
Comment (jonparrott) [inline talisman/talisman.py]: s/reqires/requires, and you can do away with the comma. Or you can word it as `Setting content_security_policy_report_only to True also requires a URI to be specified in content_security_policy_report_uri`.

File: talisman/talisman.py [hunk 4]
```diff
@@ -178,6 +195,7 @@ def _set_frame_options_headers(self, headers):
     def _set_content_security_policy_headers(self, headers):
         headers['X-XSS-Protection'] = '1; mode=block'
         headers['X-Content-Type-Options'] = 'nosniff'
+        headers['X-Download-Options'] = 'noopen'
```
Comment (jonparrott) [inline talisman/talisman.py]: We should document this new header and make it a toggle.

**Comment by jonparrott** [comment]
Almost there!

File: README.rst (hunk 1 above)
Comment (jezdez) [inline README.rst]: D'oh!

File: talisman/talisman.py (hunk 3 above)
Comment (jezdez) [inline talisman/talisman.py]: Your text sounds better indeed :)

File: talisman/talisman.py (hunk 4 above)
Comment (jezdez) [inline talisman/talisman.py]: I added the header to the README, and will add a toggle. Do you prefer to make it disabled by default?

**Comment by jezdez** [comment]
Seems like the test failure is related to the fact that flake8 wasn't only checking the talisman source code, I've sent a fix in #2.

**Comment by jezdez** [comment]
Uh, what just happened?

**Comment by jezdez** [comment]
Sorry for the noise, I sometimes forget how GitHub's wonky PR system works

**Comment by jonparrott** [comment]
Fix merged. You can try to rebase? Just let me know when you're ready for final review.

**Comment by jezdez** [comment]
@jonparrott This should be good for a final review

**Comment by jonparrott** [comment]
Thanks for this, @jezdez! I'll try to cut a new release tomorrow.

**Comment by jezdez** [comment]
@jonparrott Any chance you could push this feature to PyPI?

**Comment by jonparrott** [comment]
@jezdez I've tagged the release but Travis is mad at us. When it does run it should upload the new version (0.2.0).

**Pull request merged**